lib/database_cleaner/active_record/truncation.rb

```diff
@@ -242,7 +242,14 @@ def clean
 
     def tables_to_truncate(connection)
       tables_in_db = cache_tables? ? connection.database_cleaner_table_cache : connection.tables
-      (@only || tables_in_db) - @tables_to_exclude - connection.database_cleaner_view_cache
+      to_reject = (@tables_to_exclude + connection.database_cleaner_view_cache)
+      (@only || tables_in_db).reject do |table|
+          if table.match(/([^.]+)$/)
+            to_reject.include?($1)
```

Comment:
Please do not use `$1`. These types of variables are too cryptic for most developers and there are always ways around them. In this case, please assign the result from `table.match(/([^.]+)$/)` to a local variable and use that one.